File: Tests/src/pages/BasePage.java
package pages;

import java.util.ArrayList;
import java.util.List;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

public class BasePage extends PageWaits {
	public BasePage(WebDriver driver) {
		super(driver);
	}
	
	public String getTitle() {
		return driver.getTitle();
	}
	
	protected String getSelectValue(WebElement element) {
		Select select = new Select(element);
		return select.getFirstSelectedOption().getText();
	}
	
	protected void setSelectValue(WebElement element, String value) {
		Select select = new Select(element);
		select.selectByVisibleText(value);
	}
	
	protected List<String> getSelectOptions(WebElement element){
		Select select = new Select(element);
		List<String> result = new ArrayList<String>();
		
		for (WebElement option : select.getOptions()) {
			result.add(option.getText());
		}
		
		return result;
	}
	
	protected String getInputValue(WebElement element) {
		return element.getAttribute("value");
	}

	protected void setInputValue(WebElement element, String value) {
		element.clear();
		element.sendKeys(value);
	}
}

File: Tests/pages/BasePage.py
from selenium.common.exceptions import NoSuchElementException
from selenium.webdriver.support.select import Select
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class BasePage(object):
    def __init__(self, driver):
        self.driver = driver

    def element_exists(self, *locator):
        try:
            self.driver.find_element(*locator)
        except NoSuchElementException:
            return False
        return True

    def find_element(self, *locator): return self.driver.find_element(*locator)
    def find_elements(self, *locator): return self.driver.find_elements(*locator)

    def get_select(self, select_id): return Select(self.find_element(By.ID, select_id))
    def get_selected_option(self, select_id): return self.get_select(select_id).first_selected_option.text
    def get_select_options(self, select_id): return [option.text for option in self.get_select(select_id).options]
    def select_in_dropdown(self, select_id, value): self.get_select(select_id).select_by_visible_text(value)

    def get_input_value(self, input_id): return self.find_element(By.ID, input_id).get_attribute("value")

    def set_input_value(self, value, input_id):
        control = self.find_element(By.ID, input_id)
        control.click()
        control.clear()
        control.send_keys(value)

    def wait(self, seconds): return WebDriverWait(self.driver, seconds)
    def wait_for_clickable(self, seconds, mark):  self.wait(seconds).until(EC.element_to_be_clickable(mark))

    def wait_for_alert(self, seconds):
        try:
            self.wait(seconds).until(EC.alert_is_present())
            return self.driver.switch_to.alert
        except Exception:
            return None

    def switch_to_frame(self, frame_id):
        self.driver.switch_to.frame(self.find_element(By.ID, frame_id))

    def switch_to_default_frame(self):
        self.driver.switch_to.default_content()
